WIT_Smile_Detector.ipynb

```diff
@@ -209,17 +212,33 @@
         "\n",
         "# Setup the tool with the test examples and the trained classifier\n",
         "config_builder = WitConfigBuilder(examples[:num_datapoints]).set_custom_predict_fn(custom_predict)\n",
+        "\n",
+        "# Decode an image from tf.example bytestring\n",
+        "def decode_image(ex):\n",
+        "  im_bytes = ex.features.feature['image/encoded'].bytes_list.value[0]\n",
+        "  im = Image.open(BytesIO(im_bytes))\n",
+        "  return im\n",
+        "\n",
+        "# Define the custom distance function that compares the average color of images\n",
+        "def image_mean_distance(ex, exs, params):\n",
+        "  selected_im = decode_image(ex)\n",
+        "  mean_color = np.mean(selected_im, axis=(0,1))\n",
+        "  image_distances = [np.linalg.norm(mean_color - np.mean(decode_image(e), axis=(0,1))) for e in exs]\n",
+        "  return image_distances\n",
+        "\n",
+        "config_builder = config_builder.set_custom_distance_fn(image_mean_distance)\n",
```

Comment:
simpler code to define this above and then set it in the WitConfigBuilder line above

Reply:
done.